tools/db_bench_tool.cc

```diff
@@ -5671,17 +5681,39 @@ class Benchmark {
       }
       Status s;
       pinnable_val.Reset();
+      for (size_t i = 0; i < pinnable_vals.size(); ++i) {
+        pinnable_vals[i].Reset();
+      }
+      ColumnFamilyHandle* cfh;
       if (FLAGS_num_column_families > 1) {
-        s = db_with_cfh->db->Get(options, db_with_cfh->GetCfh(key_rand), key,
-                                 &pinnable_val, ts_ptr);
+        cfh = db_with_cfh->GetCfh(key_rand);
       } else {
-        s = db_with_cfh->db->Get(options,
-                                 db_with_cfh->db->DefaultColumnFamily(), key,
-                                 &pinnable_val, ts_ptr);
+        cfh = db_with_cfh->db->DefaultColumnFamily();
+      }
+      if (read_operands_) {
+        GetMergeOperandsOptions get_merge_operands_options;
+        get_merge_operands_options.expected_max_number_of_operands =
+            static_cast<int>(pinnable_vals.size());
+        int number_of_operands;
+        s = db_with_cfh->db->GetMergeOperands(
+            options, cfh, key, pinnable_vals.data(),
+            &get_merge_operands_options, &number_of_operands);
+        if (s.IsIncomplete()) {
+          // OK if it happens a few times.
+          pinnable_vals.resize(number_of_operands);
+          s = Status::OK();
+          continue;
+        }
+      } else {
+        s = db_with_cfh->db->Get(options, cfh, key, &pinnable_val, ts_ptr);
       }
+
       if (s.ok()) {
         found++;
         bytes += key.size() + pinnable_val.size() + user_timestamp_size_;
+        for (size_t i = 0; i < pinnable_vals.size(); ++i) {
+          bytes += pinnable_vals[i].size();
+        }
```

Comment:
Wouldn't it be more efficient to loop on number_of_operands?

Reply:
I thought so too. What stopped me though is the API doc on GetMergeOperands() only claims `number_of_operands` is set when it returns `Status::Incomplete`: https://github.com/facebook/rocksdb/blob/de9df6e818774998c50e715c73736531ea42ad44/include/rocksdb/db.h#L552. That seems dubious, but I didn't want to challenge the API contract here.

Reply:
My guess is that the documentation would be incorrect.  Otherwise, how do you know how many values are actually returned?  How would you know if the DB contained a bunch of empty values?  You would always need to loop through whatever size array was allocated as one of them somewhere might not be empty, which would be odd and very inefficient.

Reply:
I'd guess that too, or the function is buggy for empty values. But neither case is within the scope of this PR.